File: courzelo-dashboard-LMS-main/src/app/shared/models/Revision/Revision.ts
import { FileMetadatarevision } from "./FileMetadatarevision";
import { QuizRevision } from "./QuizzRevisison/QuizRevision ";
import { subjectRevision } from "./SubjectRevision";

export interface Revision {

       id: string; 
      titre: string;
      nbrmax: number;
      subjectRevision: subjectRevision;
      quizzRevisions?: QuizRevision[];
      files?: FileMetadatarevision[];
      
    }

File: Courzelo-backend-LMS/Courzelo/src/main/java/org/example/courzelo/models/RevisionEntities/revision/Revision.java
package org.example.courzelo.models.RevisionEntities.revision;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.courzelo.models.RevisionEntities.QizzRevision.QuizRevision;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "Revision")
public class Revision {
    @Id
    private String id;

    @Indexed
    private String titre;

    @DBRef
    private List<FileMetadatarevision> files;

    private int nbrmax;

    @Indexed
    private SubjectRevision subjectRevision;

    @DBRef
    private List<QuizRevision> quizRevisions;
}